source/backend/dal/Repositories/CompReqFinancialRepository.cs

```diff
@@ -62,48 +64,73 @@ public IEnumerable<PimsCompReqFinancial> GetAllByLeaseFileId(long leaseFileId, b
+                query = query
+                    .Include(f => f.CompensationRequisition)
+                        .ThenInclude(cr => cr.AcquisitionFile)
+                            .ThenInclude(a => a.PimsAcquisitionFileTeams)
+                                .ThenInclude(afp => afp.Person)
+                    .Include(f => f.CompensationRequisition)
+                        .ThenInclude(cr => cr.AcquisitionFile)
+                            .ThenInclude(a => a.PimsAcquisitionFileTeams)
+                                .ThenInclude(o => o.Organization)
+                    .Include(f => f.CompensationRequisition)
+                        .ThenInclude(cr => cr.AcquisitionFile)
+                            .ThenInclude(a => a.Project)
+                    .Include(f => f.CompensationRequisition)
+                        .ThenInclude(cr => cr.AcquisitionFile)
+                            .ThenInclude(a => a.Product);
+            }
+
+            if (includeLeases)
+            {
+                query = query
+                    .Include(f => f.CompensationRequisition)
+                        .ThenInclude(cr => cr.Lease)
+                            .ThenInclude(l => l.Project)
+                    .Include(f => f.CompensationRequisition)
+                        .ThenInclude(cr => cr.Lease)
+                            .ThenInclude(l => l.Product);
+            }
+
+            var predicate = PredicateBuilder.New<PimsCompReqFinancial>(p => true);
 
             if (filter.Projects != null && filter.Projects.Any())
             {
-                query = query.Where(f =>
-                    (f.CompensationRequisition.AlternateProjectId.HasValue && filter.Projects.Contains(f.CompensationRequisition.AlternateProjectId.Value)) ||
-                    (!f.CompensationRequisition.AlternateProjectId.HasValue && f.CompensationRequisition.AcquisitionFile.ProjectId.HasValue && filter.Projects.Contains(f.CompensationRequisition.AcquisitionFile.ProjectId.Value)));
+                var projectBuilder = PredicateBuilder.New<PimsCompReqFinancial>(p => false);
+                projectBuilder.Or(f => f.CompensationRequisition.AlternateProjectId.HasValue && filter.Projects.Contains(f.CompensationRequisition.AlternateProjectId.Value));
+                projectBuilder.Or(f => !f.CompensationRequisition.AlternateProjectId.HasValue && f.CompensationRequisition.AcquisitionFile != null && f.CompensationRequisition.AcquisitionFile.ProjectId.HasValue && filter.Projects.Contains(f.CompensationRequisition.AcquisitionFile.ProjectId.Value));
+                projectBuilder.Or(f => !f.CompensationRequisition.AlternateProjectId.HasValue && f.CompensationRequisition.Lease != null && f.CompensationRequisition.Lease.ProjectId.HasValue && filter.Projects.Contains(f.CompensationRequisition.Lease.ProjectId.Value));
+
+                predicate.And(projectBuilder);
             }
 
             if (filter.AcquisitionTeamPersons != null && filter.AcquisitionTeamPersons.Any())
```

Comment:
it could probably be usefull to include the 'includeAcquisitions' in this check since TeamPersons only applies to Acquisition Files.

Reply:
good catch